lua/lspsaga/provider.lua

```diff
@@ -500,7 +500,13 @@ function lspfinder.preview_definition(timeout_ms)
   if vim.tbl_islist(result) then
     for _, definition in ipairs(result) do
       if not vim.tbl_isempty(definition) and not vim.tbl_isempty(definition.result) then
-        local uri = definition.result[1].uri or definition.result[1].targetUri
+        local definition_result = nil
+        if definition.result[1] then
+          definition_result = definition_result[1]
+        else
+          definition_result = definition.result
+        end
+        local uri = definition_result.uri or definition_result.targetUri
```

Comment:
Did some testing on rust server and lua server: got the following error

`provider.lua:505: attempt to index local 'definition_result' (a nil value)`


`result` after `vim.tbl_islist` in rust server:

```lua
{ {
    result = { {
        originSelectionRange = {
          end = {
            character = 36,
            line = 5
          },
          start = {
            character = 19,
            line = 5
          }
        },
        targetRange = {
          end = {
            character = 1,
            line = 91
          },
          start = {
            character = 0,
            line = 40
          }
        },
        targetSelectionRange = {
          end = {
            character = 24,
            line = 51
          },
          start = {
            character = 7,
            line = 51
          }
        },
        targetUri = "file:///Users/tami5/sources/mac-notification-sys/src/lib.rs"
      } }
  } }
```


`result` after `vim.tbl_islist` in lua server:

```lua
{ {
    result = { {
        range = {
          end = {
            character = 37,
            line = 483
          },
          start = {
            character = 19,
            line = 483
          }
        },
        uri = "file:///Users/tami5/.local/share/nvim/site/pack/packer/start/lspsaga.nvim/lua/lspsaga/provider.lua"
      } }
  } }
```

Reply:
oops sorry, it was a typo, I've committed a fix, could you test again? I tried with both your examples and works fine